libsolidity/parsing/Token.cpp

```diff
@@ -113,7 +113,7 @@ int Token::parseSize(string::const_iterator _begin, string::const_iterator _end)
 {
 	try
 	{
-		unsigned int m = boost::lexical_cast<int>(boost::make_iterator_range(_begin, _end));
+		unsigned int m = boost::lexical_cast<unsigned int>(boost::make_iterator_range(_begin, _end));
```

Comment:
I think lexical-casting it to `int` is better (we check that it does not contain `-` actually), because we also return int. Some compilers might complain about the conversion. You could just use `return boost::lexical_cast<int>(...)`.